src/ILCompiler.CppCodeGen/src/ILCompiler.CppCodeGen.csproj

```diff
@@ -30,6 +30,9 @@
     <Compile Include="..\..\Common\src\Internal\Runtime\RuntimeConstants.cs">
       <Link>Runtime\RuntimeConstants.cs</Link>
     </Compile>
+    <Compile Include="..\..\Test.CoreLib/src/System/Runtime/CompilerServices/StaticClassConstructionContext.cs">
+      <Link>System/Runtime/CompilerServices/StaticClassConstructionContext.cs</Link>
```

Comment:
This file should not be included in the compiler. It would produce wrong result, e.g. when the compiler is running on 64-bit machine, but producing 32-bit output.

You should compute the size by emitting `sizeof(...)` in the C++ output (it is what the C++ codegen typically does) or asking the type system to get the size of the type.

Reply:
Thank you!